e2e/e2eutil/utils.go

```diff
@@ -100,3 +110,26 @@ func WaitForAllocRunning(t *testing.T, nomadClient *api.Client, allocID string)
 		t.Fatalf("failed to wait on alloc: %v", err)
 	})
 }
+
+func WaitForDeployment(t *testing.T, nomadClient *api.Client, deployID string, status string, statusDesc string) {
+	testutil.WaitForResultRetries(retries, func() (bool, error) {
+		time.Sleep(time.Millisecond * 100)
+		deploy, _, err := nomadClient.Deployments().Info(deployID, nil)
+		if err != nil {
+			return false, err
+		}
+
+		if deploy.Status == status && deploy.StatusDescription == statusDesc {
```

Comment:
These checks feel a bit redundant?  I think Status <-> StatusDescription is 1:1 relationship but not 100% sure.

Reply:
they're not 1:1, in kind of a code smelly way, I think. It's probably worth improving if there's some future significant refactoring. There's runtime code that depends on the value of desc too.

Reply:
Thanks - let's add a ticket for that or a comment in structs.go to point that out in a follow up.